src/main/java/net/dries007/tfc/objects/te/TEBarrel.java

```diff
@@ -259,8 +259,10 @@ public void update()
                         surplus.addAll(output);
                         IBlockState state = world.getBlockState(pos);
                         world.notifyBlockUpdate(pos, state, state, 3);
+                        onSealed(); //run the sealed check again in case we have a new valid recipe.
                     }
-                    recipe = null;
+                    else recipe = null;
```

Comment:
Eww
```suggestion
                    else 
                    {
                        recipe = null;
                    }
                        
```

Reply:
This one didn't get caught by Code Format. Thanks.